src/main/java/net/minecraftforge/common/crafting/CraftingHelper.java

```diff
@@ -582,20 +584,17 @@ public static void loadRecipes()
         //TODO: Figure out how to remove recipes, and override them. This relies on cpw to help.
         //For now this is only done one after mod init, I want to move this to ServerInit and re-do it many times.
         init();
+        ForgeRegistry<IRecipe> reg = (ForgeRegistry<IRecipe>)ForgeRegistries.RECIPES;
         if (DEBUG_LOAD_MINECRAFT)
-        {
-            Iterator<IRecipe> itr = GameData.getRecipeRegistry().iterator();
-            while(itr.hasNext())
-            {
-                itr.next();
-                itr.remove();
-            }
-        }
+            reg.clear();
+        else
+            GameData.revert(RegistryManager.FROZEN, GameData.RECIPES, false);
```

Comment:
wouldn't be any major issues as far as i see no, it's just that it would mean reading them all from disk again and it with modpacks this could easely be 100-200+ recipe files that would need to be parsed again while the data is already there (unless i'm missing something here)